library/src/containers/Servers/ServerSecurity.tsx

```diff
@@ -8,82 +8,165 @@ import { ServerHelpers } from '../../helpers';
+  securitySchema: SecurityScheme | null;
+  protocol: string;
 }
 
 const ServerSecurityItem: React.FunctionComponent<ServerSecurityItemProps> = ({
   securitySchema,
+  protocol,
 }) => {
   const schemas: React.ReactNodeArray = [];
-  if (securitySchema.name()) {
+  if (securitySchema && securitySchema.name()) {
     schemas.push(<span>Name: {securitySchema.name()}</span>);
   }
-  if (securitySchema.in()) {
+  if (securitySchema && securitySchema.in()) {
     schemas.push(<span>In: {securitySchema.in()}</span>);
   }
-  if (securitySchema.scheme()) {
+  if (securitySchema && securitySchema.scheme()) {
     schemas.push(<span>Scheme: {securitySchema.scheme()}</span>);
   }
-  if (securitySchema.bearerFormat()) {
+  if (securitySchema && securitySchema.bearerFormat()) {
     schemas.push(<span>Bearer format: {securitySchema.bearerFormat()}</span>);
   }
-  if (securitySchema.openIdConnectUrl()) {
+  if (securitySchema && securitySchema.openIdConnectUrl()) {
     schemas.push(
       <Href href={securitySchema.openIdConnectUrl()} className="underline">
         Connect URL
       </Href>,
     );
   }
 
-  const flows = securitySchema.flows();
+  let renderedKafkaSecurity;
+  if (protocol === 'kafka' || protocol === 'kafka-secure') {
+    let securityProtocol;
+    let saslMechanism;
+    if (protocol === 'kafka') {
+      if (securitySchema) {
+        securityProtocol = 'SASL_PLAINTEXT';
+      } else {
+        securityProtocol = 'PLAINTEXT';
+      }
+    } else {
+      if (securitySchema) {
+        securityProtocol = 'SASL_SSL';
+      } else {
+        securityProtocol = 'SSL';
+      }
+    }
+    if (securitySchema) {
+      switch (securitySchema.type()) {
+        case 'plain':
+          saslMechanism = 'PLAIN';
+          break;
+        case 'scramSha256':
+          saslMechanism = 'SCRAM-SHA-256';
+          break;
+        case 'scramSha512':
+          saslMechanism = 'SCRAM-SHA-512';
+          break;
+        case 'oauth2':
+          saslMechanism = 'OAUTHBEARER';
+          break;
+        case 'gssapi':
+          saslMechanism = 'GSSAPI';
+          break;
+        case 'X509':
+          securityProtocol = 'SSL';
+          break;
+      }
+    }
+
+    renderedKafkaSecurity = (
+      <div className="px-4 py-2 ml-2 mb-2 border border-gray-400 bg-gray-100 rounded">
+        {securityProtocol && (
+          <div className="mt-1">
+            <span className="text-xs text-gray-600 mt-1 mr-1">
+              security.protocol:
+            </span>
+            <span className="text-xs text-gray-600 mt-1 mr-1 uppercase">
+              {securityProtocol}
+            </span>
+          </div>
+        )}
+        {saslMechanism && (
+          <div className="mt-1">
+            <span className="text-xs text-gray-600 mt-1 mr-1">
+              sasl.mechanism:
+            </span>
+            <span className="text-xs text-gray-600 mt-1 mr-1 uppercase">
+              {saslMechanism}
+            </span>
```

Comment:
To be consistent with other styling I suggest to render such a fields with styling like for `flows` properties, so please propagate my suggestions:

 ```suggestion
            <span className="text-xs font-bold text-gray-600 mt-1 mr-1 uppercase">
              security.protocol:
            </span>
            <span className="inline-block font-bold no-underline bg-indigo-400 text-white text-xs rounded py-0 px-1 ml-1">
              {securityProtocol}
            </span>
          </div>
        )}
        {saslMechanism && (
          <div className="mt-1">
            <span className="text-xs font-bold text-gray-600 mt-1 mr-1 uppercase">
              sasl.mechanism:
            </span>
            <span className="inline-block font-bold no-underline bg-indigo-400 text-white text-xs rounded py-0 px-1 ml-1">
              {saslMechanism}
            </span>
```

and then it will be renderer like:

![image](https://user-images.githubusercontent.com/20404945/135827074-cc6a7f9c-7a2d-4192-993b-a02e3f227704.png)